src/Surfnet/StepupMiddleware/ApiBundle/Identity/Repository/AuthorizationRepository.php

```diff
@@ -82,6 +82,43 @@ public function getInstitutionsForRole(InstitutionRole $role, IdentityId $actorI
         return $result;
     }
 
+    /**
+     * Finds the institutions that have the Select RAA authorizatio based on
+     * the institution of the specified identity.
+     */
+    public function getInstitutionsForSelectRaaRole(IdentityId $actorId)
+    {
+        $qb = $this->_em->createQueryBuilder()
+            ->select("ci.institution")
+            ->from(InstitutionAuthorization::class, 'ia')
+            ->innerJoin(ConfiguredInstitution::class, 'ci', Join::WITH, 'ia.institutionRelation = ci.institution')
+            ->innerJoin(Identity::class, 'i', Join::WITH, 'ia.institution = i.institution AND i.id = :identityId')
```

Comment:
Is the inner join on Identity::class required?

Reply:
Yeah, we want to filter on the identity id of the identity, where we join on the institution. Or do you see another way to achieve this?

Reply:
I reckoned that the join on the IdentityId of the RaListing would have been sufficient, but feel free to leave it as-is.